### PR TITLE
Initial PR for RAT License Check 

### PR DESCRIPTION
**Pull request by mbaijal**
## Description ##
Goal of this PR - 
1. To get the .rat-excludes file into the 1.2 Release so that RAT check can be run locally during src code validation (voting process)
2. Get the .rat-excludes file reviewed by everyone before it is merged.

Please let me know if you see any file/directory or file format that should not be included in this list or if there are any regex errors. I have been adding incrementally to this list and a redundant line may have creeped in.

refer this link for further details - https://cwiki.apache.org/confluence/display/MXNET/MXNet+Source+Licenses

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added a directory for rat check nightly test (script coming soon but may be included after 1.2)
- [x] Added a README.md for rat check
- [x] Added a .rat-excludes file

## Comments ##
@gautamkmr @eric-haibin-lin @bhavinthaker 
Please Review

### CODE REVIEW

File: tests/nightly/apache_rat_license_check/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ 
+### The .rat-excludes file
+This file lists all the files, directories and file formats that are excluded from license checks for various reasons.
+If you think something is wrong, feel free to change!
+
+### Nightly test script for license check
+Coming soon...
+
+### How to run the RAT check locally
+The following commands can be used to run a Apache RAT check locally -
```
Comment (marcoabreu) [inline tests/nightly/apache_rat_license_check/README.md]: Would it be possible to incorporate this with our ci-docker setup in order to allow people to execute a one-liner without requiring installation of any local dependencies?

File: tests/nightly/apache_rat_license_check/.rat-excludes [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+3rdparty/*
+R-package/*
+src/operator/mkl/*
+trunk/*
+docker/*
+docker_multiarch/*
+.*\\.m
+.*\\.mk
+.*\\.R
+contrib/*
```
Comment (eric-haibin-lin) [inline tests/nightly/apache_rat_license_check/.rat-excludes]: the contrib folder is ignored?

File: tests/nightly/apache_rat_license_check/.rat-excludes [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Dockerfile*
+.*svg
+.*cfg
+.*config
+docs/*
+__init__.py
+build/*
+.*\\.t
+perl-package/*
```
Comment (eric-haibin-lin) [inline tests/nightly/apache_rat_license_check/.rat-excludes]: Ignoring the entire perl-package?

File: tests/nightly/apache_rat_license_check/.rat-excludes [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+coco.py
+base.pyi
+special_functions-inl.h
+im2col.cuh
+im2col.h
+pool.h
+README.rst
+dataset.cPickle
+
+rcnn/*
```
Comment (eric-haibin-lin) [inline tests/nightly/apache_rat_license_check/.rat-excludes]: Why excluding these two examples?

File: tests/nightly/apache_rat_license_check/README.md (hunk 1 above)
Comment (mbaijal) [reply tests/nightly/apache_rat_license_check/README.md]: Yes, a script is coming with the nightly tests

File: tests/nightly/apache_rat_license_check/.rat-excludes (hunk 3 above)
Comment (mbaijal) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: Yea, these are packages I am not sure about the licensing, I have added details here - 
https://cwiki.apache.org/confluence/display/MXNET/MXNet+Source+Licenses

File: tests/nightly/apache_rat_license_check/.rat-excludes (hunk 2 above)
Comment (mbaijal) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: I dont think contrib can have the apache header. 
Again, I am not sure what the best way to handle these is

File: tests/nightly/apache_rat_license_check/README.md (hunk 1 above)
Comment (marcoabreu) [reply tests/nightly/apache_rat_license_check/README.md]: Great :)

**Comment by anirudh2290** [comment]
@mbaijal any updates ?

**Comment by mbaijal** [comment]
I think this PR can be merged as is. The list of directories included in the .rat-excludes file has been discussed in the previous release and reasons for most decisions are documented here - https://cwiki.apache.org/confluence/display/MXNET/MXNet+Source+Licenses

In most cases, either the file should have no license or I am unsure about the correct license.
As we move forward, where ever possible the community can add the correct licenses and clean this list further.

File: tests/nightly/apache_rat_license_check/.rat-excludes (hunk 2 above)
Comment (eric-haibin-lin) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: How does other apache projects handle contrib folder? I thought licenses are required

File: tests/nightly/apache_rat_license_check/.rat-excludes (hunk 3 above)
Comment (eric-haibin-lin) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: @sergeykolychev any comment about adding apache license to these files?

Comment (sergeykolychev) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: I'll take care of these

**Comment by eric-haibin-lin** [comment]
I'm going to merge this for now. 

@sergeykolychev please update the perl license when you get time. I appreciate it.

@mbaijal thanks for adding the rat-check.

File: tests/nightly/apache_rat_license_check/.rat-excludes (hunk 2 above)
Comment (mbaijal) [reply tests/nightly/apache_rat_license_check/.rat-excludes]: @eric-haibin-lin, see my comment at the end of this PR. 
I do not want to add wrong licenses to any files and hence have left these files as is until I am sure they can have the ASF header.

**Pull request merged**